posixsrv.c

```diff
@@ -288,7 +289,15 @@ void posixsrv_threadMain(void *arg)
 			if (o != NULL) {
 				posixsrv_object_put(o);
 			}
-			r->msg.o.err = -EINVAL;
+
+			/* Make isatty() set correct errno */
+			if (r->msg.type == mtDevCtl && ((ioctl_in_t *)r->msg.i.raw)->request == TCGETS) {
```

Comment:
1. please add correct handling for special files only (not in ultra-generic API entry point)
2. please use `ioctl_unpack` && `ioctl_setResponse` (this should be the only way to interact with ioctl data)